src/main/kotlin/test/ktortemplate/core/utils/json/PageResponseDeserializer.kt

```diff
@@ -0,0 +1,31 @@
+package test.ktortemplate.core.utils.json
+
+import com.google.gson.JsonDeserializationContext
+import com.google.gson.JsonDeserializer
+import com.google.gson.JsonElement
+import com.google.gson.JsonParseException
+import test.ktortemplate.core.utils.pagination.PageResponse
+import test.ktortemplate.core.utils.pagination.PageResponseLink
+import test.ktortemplate.core.utils.pagination.PageResponseMeta
+import java.lang.reflect.Type
+import java.util.ArrayList
+
+class PageResponseDeserializer<T>(private val clazz: Class<T>) : JsonDeserializer<PageResponse<T>> {
```

Comment:
Gson does not handle well the deserialisation of generic types such as PageResponse<T>, so a custom deserialiser must be written.

However, this does not affect serialisation which means that serialised classes are ok, and the jsons returned by the endpoint are the same.
Although, for the tests, the custom deserialiser is necessary to convert from : `PageResponse(meta=PageResponseMeta(page=0, size=10, totalElements=5), data=[{id=1.0, brand=brand10, model=model91}, {id=2.0, brand=brand72, model=model12}…)`
to `PageResponse(meta=PageResponseMeta(page=0, size=10, totalElements=5), data=[Car(id=1, brand=brand10, model=model91), Car(id=2, brand=brand72, model=model12)…)`

Reply:
I don't this this deserializer is needed, this sample works fine:

```kotlin
val sample = """
            {
	"meta": {"page":0, "size":0, "totalElements":0},
	"links": {"self":"", "first":"", "last": ""},
	"links": {"self":"", "first":"", "last": ""},
	"data": [{
		"id": 1.0,
		"brand": "brand10",
		"model": "model91"
	}, {
		"id": 2.0,
		"brand": "brand72",
		"model": "model12"
	}]
}
        """.trimIndent()

val response = Gson().fromJson<PageResponse<*>>(sample, object : TypeToken<PageResponse<Car>>() {}.type)

println(response)

// Result
PageResponse(meta=PageResponseMeta(page=0, size=0, totalElements=0), data=[Car(id=1, brand=brand10, model=model91), Car(id=2, brand=brand72, model=model12)], links=PageResponseLink(self=, first=, prev=null, next=null, last=))
```

>